### ISSUE TITLE
Nullness Checker misses explicit @Nullable return of ASTHelpers.getSymbol (bytecode dependency)

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#1992]
`com.google.errorprone.util.ASTHelpers.getSymbol(ClassTree)` has an explicit jspecify `@Nullable` return, present in the shipped bytecode:

```
public static com.sun.tools.javac.code.Symbol$ClassSymbol getSymbol(com.sun.source.tree.ClassTree);
  RuntimeVisibleTypeAnnotations:
    org.jspecify.annotations.Nullable
```

Yet the Nullness Checker does not report a dereference of its result. Reproduces with a **bare** `NullnessChecker` invocation — no `-AskipUses`, no `-AconservativeArgumentNullnessAfterInvocation`, no `@NullMarked` involvement on the client side.

`ClientReal.java`:

```java
import com.google.errorprone.util.ASTHelpers;
import com.sun.source.tree.ClassTree;
import com.sun.tools.javac.code.Symbol.ClassSymbol;

public class ClientReal {
    void use(ClassTree tree) {
        ClassSymbol s = ASTHelpers.getSymbol(tree);
        s.getNestingKind(); // expected dereference.of.nullable; NONE reported
    }
}
```

```
checker/bin/javac -processor org.checkerframework.checker.nullness.NullnessChecker \
  -cp "<error_prone_check_api-2.50.0.jar>:<jspecify-1.0.1.jar>:checker/dist/checker-qual.jar:<guava.jar>" \
  --add-exports jdk.compiler/com.sun.source.tree=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
  -proc:only ClientReal.java
```

→ no warning.

**Ruled out:**

- `-AskipUses=com.sun.*` — same result (0 warnings) with and without it.
- `@NullMarked` package alone — a hand-compiled `@Nullable` method in a `@NullMarked` package **is** correctly flagged.
- Explicit `@Nullable` in bytecode in general — hand-compiled control libraries with an explicit jspecify (or Checker Framework) `@Nullable` return, including `com.sun.*` return types, **are** correctly flagged.

So the miss is specific to the real `error_prone_check_api` `ASTHelpers.getSymbol`. Suspects to investigate: the overloaded `getSymbol(Tree)` / `getSymbol(ClassTree)` pair, the generic signature, or how Error Prone emits its jspecify type-use annotations into bytecode.

**Impact:** the `framework-errorprone` module is dog-fooded under `checkNullness`, but null Error Prone API returns like this slip through. This surfaced a real NPE risk in `EisopCheckerFrameworkPlugin.matchClass` (dereferencing `getSymbol(tree)` without a null check), which is fixed separately with an explicit null guard.

Found while adding the Error Prone plugin (branch `cf-ep`).
